Provenance-Emu/Provenance
# Cheats: Update Play, Mednafen, SNES9X with Code Type Support

### What does this PR do
   This updates the Mednafen and SNES9X Cores with cheat code capability with specific code type, giving a more accurate result. This PR has:
   - Update to SNES9X Cheat with support for Game Genie, Pro Action Replay, Gold Finger, and Raw Code. Gold Finger includes X as part of the code, so cheat input is updated to support all alpha numerics
   - Update to Mednafen, moves cheat to PVMednafenCore+Cheats, supports GameGenie/Pro Action Replay on SNES, Game Shark on PSX, and in addition supports GameBoy with Game Genie / Game Shark
   - Minor update to Play Core, with replacing 'Game Genie' with 'Game Shark V3' (thought decrypt was game genie, it was actually game shark)

   In addition, as part of above:
   - GameWithCheat protocol is updated to a single setCheat that receives Code, Name, Code Type, and Cheat Index (index of the cheat in the UI Table) 
   - The cores (Play / SNES9X / Mednafen) are updated with new protocol
   - The original setCheat API format is still supported (with 2nd commit), which is called if setCheat is implemented with original API format without new API format (to support all that implements the cheat feature)

### Where should the reviewer start
- SNES9X setCheat -> Mednafen+Cheats setCheat 

### How should this be manually tested
- Launch any core that supports cheat, select code type, and the cheat should be there

### Any background context you want to provide
- Adds codeType selection to existing cores with cheat feature implemented

### What are the relevant tickets
- https://github.com/Provenance-Emu/Provenance/pull/2089

### Screenshots (important for UI changes)

### Questions
